lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -124,15 +132,23 @@
         version_added: "2.7"
 
     ownca_not_before:
+        default: +0s
         description:
-            - The timestamp at which the certificate starts being valid. The timestamp is formatted as an ASN.1 TIME.
-              If this value is not specified, certificate will start being valid from now.
+            - "The point in time the certificate is valid from. Time can be specified either as relative time or as absolute timestamp.
+               Time will always be interpreted as UTC. Valid formats are: C([+-]timespec | ASN.1 TIME)
+               where timespec can be an integer + C([w | d | h | m | s]) (e.g. C(+32w1d2h).
+               Note that if using relative time this module is NOT idempotent.
+               If this value is not specified, the certificate will start being valid from now."
         version_added: "2.7"
 
     ownca_not_after:
+        default: +3650d
```

Comment:
This is exactly how the module worked up until now, 10 years are not a precisely definable time span.

Reply:
Then the documentation of the module has been wrong so far. 10 years from now is well-defined.

Reply:
Alright, what time and date is it exactly 10 years after 2012-02-29 and how many days are in ten years exactly?

Reply:
Why do you want to convert this into a number of days? Yes, I have to admit it's not totally well-defined (for years because of leap years, and for months because of variable number of days per month), but with standard rounding down it does what people expect.

Reply:
(I really shouldn't have used well-defined in the first place :) )

Reply:
These edge cases need to be considered though and documented as well - or you just let people choose for themselves if they want to define "10 years" as the average of "3652d12h", as "10 * 365 days" or as "exactly today's date in 10 years from now" (not possible for dates like the 29th of February). I prefer the latter, the documentation alone what exactly "1 month from today" means in this case would be as tedious to read as if it would be to write (using an example: One month after January 28th, 29th, 30th and 31st would be what day exactly in "normal" and leap years?).

Reply:
How about just documenting that using months and years uses potential rounding, and leaving the details away?

Maybe we should also just don't add this feature now.

Reply:
> Maybe we should also just don't add this feature now.

Do you mean (a) relative timestamps or (b) being able to define years in relative timestamps with "this feature"?

Currently `cert.gmtime_adj_notAfter(315360000)` is used by the way, which is 3650 days in seconds.

Reply:
Ah, sorry, I meant years/months support, not the general relative timestamp feature :)

Reply:
Ah, ok. Yes, I agree with that one. :-)